core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -455,8 +455,9 @@ public Path getWorkingDirectory() {
   public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throws IOException {
     Preconditions.checkArgument(uri.getScheme().equals(getScheme()),
         PreconditionMessage.URI_SCHEME_MISMATCH.toString(), uri.getScheme(), getScheme());
-    Preconditions.checkArgument(mInitialized.compareAndSet(false, true), "Cannot invoke "
-        + "initialize() more than once");
+    if (!mInitialized.compareAndSet(false, true)) {
```

Comment:
If 2 threads initialize(), the first one will succeed, and the second will return. However, If the initialize takes some time, the second thread which returned will not have an initialized client, right?

Reply:
This class is not thread safe. But I updated the PR for thread safe concurrent initialization.